named-snapshot.tmux

```diff
@@ -8,49 +8,78 @@ source "$CURRENT_DIR/scripts/variables.sh"
 bind_prompt_key() {
   local action="$1"
   local key="$2"
+  local switch_client="$4"
+  local client_name=""
+
   if [[ "$action" == "save" ]]; then
     local prompt='Save Snapshot as:'
   else
     local prompt='Restore Snapshot:'
   fi
-  tmux bind-key "$key" command-prompt -p "$prompt" "run-shell '$CURRENT_DIR/scripts/$action-snapshot.sh %1'"
+  if [ -n "$switch_client" ]; then
+    client_name="$(echo "$switch_client" | cut -d':' -f2)"
+    tmux bind-key -T "$client_name" "$key" command-prompt -p "$prompt" "run-shell '$CURRENT_DIR/scripts/$action-snapshot.sh %1'"
+  else
+    tmux bind-key "$key" command-prompt -p "$prompt" "run-shell '$CURRENT_DIR/scripts/$action-snapshot.sh %1'"
+  fi
 }
 
 bind_key() {
   local action="$1"
   local key="$2"
   local name="$3"
+  local switch_client="$4"
 
   if test "$name" = "*"; then
     bind_prompt_key "$@"
   else
-    tmux bind-key "$key" run-shell "$CURRENT_DIR/scripts/$action-snapshot.sh $name"
```

Comment:
suggestion: could you refactor this into a reusable function call?

With the above newly introduced function, we can also refactor

```sh
bind_key() {
  # ...
  local key_table="root"
  if [ -n "$switch_client" ]; then
    key_table="$(echo "$switch_client" | cut -d':' -f2)"
  fi
  bind_tmux_key "$key_table" "$key" "" "$action"
  # ...
}
```

Reply:
We could even move a check to properly override a key table to a new `bind_tmux_key` altogether to remove even more duplicate code!

Reply:
Replied here: https://github.com/spywhere/tmux-named-snapshot/pull/5#issuecomment-2207154911